src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -23,11 +23,25 @@ class FirewallContext
 {
     private $listeners;
     private $exceptionListener;
+    private $config;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null)
+    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
     {
         $this->listeners = $listeners;
         $this->exceptionListener = $exceptionListener;
+
+        if (null === $config) {
+            @trigger_error(sprintf('%s() takes an instance of %s as third argument since version 3.2 and will throw an exception if not set in 4.0.', __METHOD__, FirewallConfig::class), E_USER_DEPRECATED);
```

Comment:
I'd suggest:

```diff
- '%s() takes an instance of %s as third argument since version 3.2 and will throw an exception if not set in 4.0.'
+ '"%s()" expects an instance of "%s" as third argument since version 3.2 and will throw an exception in 4.0 if not provided.'
```

But I didn't checked for similar deprecation messages 😅